csrc/executor_kernel_arg.h

```diff
@@ -30,23 +34,27 @@ struct TensorArgCodegen {
   constexpr int nDims() const {
     return N;
   }
-  void setSize(int i, nvfuser_index_t s) {
+  void setSize(int64_t i, nvfuser_index_t s) {
     size[i] = s;
   }
-  void setStride(int i, nvfuser_index_t s) {
+  void setStride(int64_t i, nvfuser_index_t s) {
     stride[i] = s;
   }
-  nvfuser_index_t getSize(int i) const {
+  nvfuser_index_t getSize(int64_t i) const {
     return size[i];
   }
-  nvfuser_index_t getStride(int i) const {
+  nvfuser_index_t getStride(int64_t i) const {
     return stride[i];
   }
 };
 
 // 0-Dim GPU based tensor
 template <typename T, typename nvfuser_index_t>
 struct TensorArgCodegen<T, 0, nvfuser_index_t> {
+  using data_type = T;
+  using index_type = nvfuser_index_t;
```

Comment:
does `index_type` need to be consistent across all input tensors? Otherwise it feels strange that we care about `index_type` for a scalar tensor :laughing:

Reply:
Yes because `TensorArg` has the methods like below and is also instantiated with the 0-dim tensor type.

https://github.com/NVIDIA/Fuser/pull/159/files#diff-108f0d9f28a8c37e5ea372fb5eef2659152a439ec8bf90f023e791c798c4838eR229